### PR TITLE
Add tests for save_idxs and update readme to current api

### PR DESCRIPTION
**Pull request by abhigupta768**

### CODE REVIEW

**Comment by Vaibhavdixit02** [comment]
Add a test for `sample_u0` as well for all the backends, that isn't tested yet

**Comment by Vaibhavdixit02** [comment]
Looks good to me now.

@ChrisRackauckas?

File: src/dynamichmc_inference.jl [hunk 1]
```diff
@@ -28,17 +28,19 @@ Base.@kwdef struct DynamicHMCPosterior{TA,TP,TD,TT,TR,TS,TK}
...
 end
 
 function (P::DynamicHMCPosterior)(θ)
     @unpack parameters, σ = θ
     @unpack algorithm, problem, data, t, parameter_priors = P
-    @unpack σ_priors, solve_kwargs, sample_u0 = P
-    nu, T = length(problem.u0), eltype(parameters)
+    @unpack σ_priors, solve_kwargs, sample_u0, save_idxs = P
+    T = eltype(parameters)
+    nu = save_idxs === nothing ? length(problem.u0) : length(save_idxs)
```
Comment (Vaibhavdixit02) [inline src/dynamichmc_inference.jl:39]: Revert this back and let's handle the combination from a different PR.

File: src/turing_inference.jl [hunk 2]
```diff
@@ -30,7 +30,7 @@ function turing_inference(
         for i in 1:length(likelihood_dist_priors)
             σ[i] ~ likelihood_dist_priors[i]
         end
-        nu = length(prob.u0)
+        nu = save_idxs === nothing ? length(prob.u0) : length(save_idxs)
```
Comment (Vaibhavdixit02) [inline src/turing_inference.jl:33]: Same

File: README.md [hunk 3]
```diff
@@ -126,7 +137,7 @@ equation solver.
 
  bayesian_result_abc = abc_inference(prob1, Tsit5(), t, data, priors)
 ```
-### Using save_idxs for using latent variables 
+### Using save_idxs to declare observables
```
Comment (ChrisRackauckas) [inline README.md:141]: you might want text here. Describe what the problem is, how to fix it, then show code.

**Comment by Vaibhavdixit02** [comment]
@abhigupta768 for `sample_u0` to be actually usable in real problems I think it is very important to have it be compatible with `save_idxs`. Let's follow up with a PR to fix that.

**Pull request merged**